### PR TITLE
C#: Add table with equivalent Dictionary methods

### PR DESCRIPTION
**Pull request by raulsntos**
- Follow up to https://github.com/godotengine/godot/pull/71984.
- Adds a table with the list of C# methods that are equivalent to the methods in Godot's Dictionary to make it easier to users to port from GDScript and find which APIs are available.

### CODE REVIEW

**Comment by paulloz** [comment]
Would it be useful to add a precision about `Godot.Collections.Dictionary` implementing `IDictionary` somewhere (same question about arrays implementing `IArray`)?

**Comment by paulloz** [comment]
Perfect.

**Comment by YuriSizov** [comment]
Should the methods and operators in this table, both GDScript and C#, be formatted as inline code with parentheses at the end? for consistency with the rest of the article. Lines that feature text should, of course, be formatted as text when appropriate.

**Comment by raulsntos** [comment]
It matches the `StringExtensions` table, the reason is that I can't add links to text formatted as inline code. In this table there are no links but I decided to be consistent with the `StringExtensions` table.

File: tutorials/scripting/c_sharp/c_sharp_differences.rst [hunk 1]
```diff
@@ -483,6 +483,30 @@ Use ``Godot.Collections.Dictionary``.
...
+is_read_only             IsReadOnly
+keys                     Keys
+make_read_only           MakeReadOnly
+merge                    Merge
+size                     Count
+values                   Values
+operator !=              !RecursiveEqual
+operator ==              RecursiveEqual
+operator []              Dictionary[Variant] indexer, Add or TryGetValue
+
```
Comment (YuriSizov) [inline tutorials/scripting/c_sharp/c_sharp_differences.rst:509]:
```suggestion
operator []              Dictionary[Variant] indexer, Add or TryGetValue
======================= ==============================================================

```

File: tutorials/scripting/c_sharp/c_sharp_differences.rst [hunk 2]
```diff
@@ -483,6 +483,31 @@ Use ``Godot.Collections.Dictionary``.
...
+is_read_only             IsReadOnly
+keys                     Keys
+make_read_only           MakeReadOnly
+merge                    Merge
+size                     Count
+values                   Values
+operator !=              !RecursiveEqual
+operator ==              RecursiveEqual
+operator []              Dictionary[Variant] indexer, Add or TryGetValue
+======================= ==============================================================
```
Comment (YuriSizov) [inline tutorials/scripting/c_sharp/c_sharp_differences.rst:509]:
```suggestion
=======================  ==============================================================
```

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**